ElectronApp/src/renderer/components/EditableTable.tsx

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint react/prop-types: 0 */
+
+import React from 'react';
+import styled from 'styled-components';
+import { useTable, usePagination } from 'react-table';
+
+import makeData from './makeData';
+
+declare module 'react-table' {
+  interface TableOptions<D extends object> {
+    updateMyData: (rowIndex: number, columnId: string, value: any) => void;
+  }
+}
+
+const Styles = styled.div`
+  padding: 1rem;
+
+  table {
+    border-spacing: 0;
+    border: 1px solid black;
+
+    tr {
+      :last-child {
+        td {
+          border-bottom: 0;
+        }
+      }
+    }
+
+    th,
+    td {
+      margin: 0;
+      padding: 0.5rem;
+      border-bottom: 1px solid black;
+      border-right: 1px solid black;
+
+      :last-child {
+        border-right: 0;
+      }
+
+      input {
+        font-size: 1rem;
+        padding: 0;
+        margin: 0;
+        border: 0;
+      }
+    }
+  }
+`;
+
+export interface EditableCellProps {
+  value: any;
+  row: any;
+  column: any;
+  updateMyData: Function;
+}
+
+const EditableCell = ({
+  value: initialValue,
+  row: { index },
+  column: { id },
+  updateMyData,
+}: EditableCellProps) => {
+  const [value, setValue] = React.useState(initialValue);
+
+  const onChange = (e: { target: { value: any } }) => {
+    setValue(e.target.value);
+  };
+
+  const onBlur = () => {
+    updateMyData(index, id, value);
+  };
+
+  React.useEffect(() => {
+    setValue(initialValue);
+  }, [initialValue]);
+
+  return <input value={value} onChange={onChange} onBlur={onBlur} />;
```

Comment:
When we later use Mui components (correct me if I'm wrong) they could be used here and in the table instead of the html ones.